Formula/php.rb

```diff
@@ -0,0 +1,324 @@
+          #{etc}/php/#{php_version}/
+    EOS
+
+    s.join "\n"
+  end
+
+  def post_install
+    (var/"log").mkpath
+    touch var/"log/php-fpm.log"
+
+    chmod 0755, lib/"php/.channels"
+    chmod 0755, lib/"php/.channels/.alias"
+    chmod 0644, (Dir.glob(lib/"php/.channels/**/*", File::FNM_DOTMATCH).reject { |a| a =~ %r{\/\.{1,2}$} || File.directory?(a) })
+    chmod 0644, %w[php/.depdblock php/.filemap php/.depdb php/.lock]
+
+    # custom location for extensions installed via pecl
+    pecl_path = HOMEBREW_PREFIX/"lib/php/#{php_version}/pecl"
+    pecl_path.mkpath
+
+    # fix pear config to use opt paths
+    php_lib_path = opt_lib/"php"
+    {
+      "php_ini" => etc/"php/#{php_version}/php.ini",
+      "php_dir" => php_lib_path,
+      "ext_dir" => pecl_path,
+      "doc_dir" => php_lib_path/"doc",
+      "bin_dir" => opt_bin,
+      "data_dir" => php_lib_path/"data",
+      "cfg_dir" => php_lib_path/"cfg",
+      "www_dir" => php_lib_path/"htdocs",
+      "man_dir" => php_lib_path/"local/man",
+      "test_dir" => php_lib_path/"test",
+      "php_bin" => opt_bin/"php",
+    }.each do |key, value|
+      system bin/"pear", "config-set", key, value, "system"
+    end
+
+    %w[
+      ldap
+      mcrypt
+      tidy
+      imap
+      opcache
+    ].each do |e|
+      next if build.devel? && (e == "mcrypt")
+      config_path = (etc/"php/#{php_version}/conf.d/ext-#{e}.ini")
+      extension_type = (e == "opcache") ? "zend_extension" : "extension"
+      if config_path.exist?
+        inreplace config_path, /#{extension_type}=.*$/, "#{extension_type}=#{e}.so"
+      else
+        config_path.write <<-EOS.undent
+          [#{e}]
+          #{extension_type}="#{e}.so"
+        EOS
+      end
+    end
+  end
+
+  def php_version
+    version.to_s.split(".")[0..1].join(".")
+  end
+
+  plist_options :startup => true, :manual => "php-fpm"
+
+  def plist; <<-EOPLIST.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>KeepAlive</key>
+        <true/>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_sbin}/php-fpm</string>
+          <string>--nodaemonize</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>WorkingDirectory</key>
+        <string>#{var}</string>
+        <key>StandardErrorPath</key>
+        <string>#{var}/log/php-fpm.log</string>
+      </dict>
+    </plist>
+    EOPLIST
+  end
+
+  test do
+    system "#{bin}/php", "-i"
+    system "#{sbin}/php-fpm", "-t"
+    system "#{bin}/phpdbg", "-V"
+    system "#{bin}/php-cgi", "-m"
+    assert_match "php7_module", shell_output(
+      %W[
+        #{Formula["httpd"].bin}/httpd -M -C
+        'LoadModule php7_module #{HOMEBREW_PREFIX}/lib/httpd/modules/libphp7.so'
+      ].join(" "))
```

Comment:
It looks odd, but the standards say the last closing parenthesis must go on the line after.

Reply:
yeah I've changed it on my system but not pushed it yet so all good.